app/controllers/logs_controller.rb

```diff
@@ -271,19 +271,23 @@ def update
   # can be given a single id or a list of ids
   def take
```

Comment:
Absolutely. I did the clean-up so I could read/understand what was going on. Its a bit hard to test, otherwise.